airflow/ui/src/pages/Run/TaskInstances.tsx

```diff
@@ -101,13 +180,59 @@ export const TaskInstances = () => {
         limit: pagination.pageSize,
         offset: pagination.pageIndex * pagination.pageSize,
         orderBy,
+        state: filteredState === null ? undefined : [filteredState],
+        taskDisplayNamePattern: Boolean(taskDisplayNamePattern)
+          ? taskDisplayNamePattern
+          : undefined,
       },
       undefined,
       { enabled: !isNaN(pagination.pageSize) },
     );
 
   return (
-    <Box>
+    <Box pt={4}>
+      <HStack>
+        <Select.Root
+          collection={stateOptions}
+          maxW="250px"
+          onValueChange={handleStateChange}
+          value={[filteredState ?? "all"]}
+        >
+          <Select.Trigger colorPalette="blue" isActive={Boolean(filteredState)}>
+            <Select.ValueText>
+              {() =>
+                filteredState === null ? (
```

Comment:
Not being a JavaScript pro but do we need to compare with `===` against null?
I would have assumed you just can say `filteredState ? (...`?

Reply:
The pattern is used in the new UI code and is in fact from the runs list tab for dag page. Using only `filteredState ?` results in below lint error. 

```
/home/karthikeyan/stuff/python/airflow/airflow/ui/src/pages/Run/TaskInstances.tsx
  204:19  error  Unexpected nullable string value in conditional. Please handle the nullish/empty cases explicitly  @typescript-eslint/strict-boolean-expressions
```